core/controllers/suggestion.py

```diff
@@ -747,29 +749,47 @@ def get(self, target_type: str, suggestion_type: str) -> None:
         Args:
             target_type: str. The type of the suggestion target.
             suggestion_type: str. The type of the suggestion.
+
+        Raises:
+            ValueError. If limit is None for question suggestions.
         """
         assert self.user_id is not None
         assert self.normalized_request is not None
         self._require_valid_suggestion_and_target_types(
             target_type, suggestion_type)
-        limit = self.normalized_request['limit']
+        limit = self.normalized_request.get('limit')
         offset = self.normalized_request['offset']
         sort_key = self.normalized_request['sort_key']
         exploration_id = self.normalized_request.get('exploration_id')
         exp_ids = [exploration_id] if exploration_id else []
-
+        user_settings = user_services.get_user_settings(self.user_id)
+        language_code_to_filter_by = (
```

Comment:
Just so I understand: what sequence of events happens (technically) when a user changes the language in the dropdown? Does doing so change the value of user_settings.preferred_translation_language_code, and does that change always happen before the request corresponding to this handler is made? I worry that there might be a race condition here.

Reply:
Yes the value of `preferred_translation_language_code` is changed whenever user selects a language in dropdown.
And yes the change always happen before the request corresponding to this handler is made so I don't think there will be a race condition here. User has to always pick language from dropdown before clicking on "Translations" button for a lesson, which makes a call to this handler.

Reply:
OK, that sounds fine then, thanks. Perhaps add a comment above this line to explain that so a reader knows how the "preferred language code" is determined.

Reply:
Done